### PR TITLE
Fix NSMapTable tests to match reference platform behavior

### PR DESCRIPTION
**Pull request by mnithish**

### CODE REVIEW

**Comment by mnithish** [comment]
@ms-jihua is added to the review. #Closed

**Comment by mnithish** [comment]
@rajsesh-msft is added to the review. #Closed

**Comment by DHowett-MSFT** [comment]
:shipit:

File: tests/unittests/Foundation/NSMapTableTests.mm [hunk 1]
```diff
@@ -80,12 +82,19 @@ void assertMapTablesEqual(NSMapTable* table1, NSMapTable* table2) {
...
     ASSERT_EQ(nil, [mapTable objectForKey:objKey]);
 
     [objKey release];
 }
 
-TEST(NSMapTable, WeakKey) {
+// Note: Test disabled on OSX
+// https://developer.apple.com/library/mac/releasenotes/Foundation/RN-FoundationOlderNotes/
+// NSMapTable zeroing weak changes - "weak-to-strong NSMapTables are not currently recommended, as the strong values for weak keys
+// which get zero'd out do not get cleared away (and released) until/unless the map table resizes itself"
```
Comment (ms-jihua) [inline tests/unittests/Foundation/NSMapTableTests.mm]: ! Did not see this at the time of implementation, apologies for making you go through this.

**Comment by ms-jihua** [comment]
:shipit:

File: tests/unittests/Foundation/NSMapTableTests.mm (hunk 1 above)
Comment (mnithish) [inline tests/unittests/Foundation/NSMapTableTests.mm]: Not a problem

---
In reply to: [73383769](https://github.com/Microsoft/WinObjC/pull/797#discussion_r73383769) [](ancestors = 73383769)

**Pull request merged**